diff --git a/core/dep/_google-webrtc/src/call/payload_type_picker.cc b/core/dep/_google-webrtc/src/call/payload_type_picker.cc
index be3885d..229aa18 100644
--- a/core/dep/_google-webrtc/src/call/payload_type_picker.cc
+++ b/core/dep/_google-webrtc/src/call/payload_type_picker.cc
@@ -369,7 +369,7 @@
 RTCError RtpHeaderExtensionRecorder::AddMapping(int id,
                                                 absl::string_view uri,
                                                 bool encrypt) {
-  auto it = uri_to_id_.find(std::pair{uri, encrypt});
+  auto it = uri_to_id_.find(std::pair<std::string, bool>{std::string(uri), encrypt});
   if (it != uri_to_id_.end()) {
     if (it->second != id) {
       RTC_HISTOGRAM_BOOLEAN(
@@ -390,7 +390,7 @@
 
 RTCErrorOr<int> RtpHeaderExtensionRecorder::LookupId(absl::string_view uri,
                                                      bool encrypt) const {
-  auto it = uri_to_id_.find(std::pair{uri, encrypt});
+  auto it = uri_to_id_.find(std::pair<std::string, bool>{std::string(uri), encrypt});
   if (it == uri_to_id_.end()) {
     return RTCError(RTCErrorType::INVALID_PARAMETER,
                     "No ID found for extension");
diff --git a/core/dep/_google-webrtc/src/third_party/libvpx/BUILD.gn b/core/dep/_google-webrtc/src/third_party/libvpx/BUILD.gn
--- a/core/dep/_google-webrtc/src/third_party/libvpx/BUILD.gn
+++ b/core/dep/_google-webrtc/src/third_party/libvpx/BUILD.gn
@@ -373,7 +373,7 @@
   # SVE is disabled for Windows due to a limitation with clang-cl-18:
   # third_party\llvm-build\Release+Asserts\lib\clang\18\include\arm_sve.h(271,1):
   # error: cannot mangle this built-in __SVInt8_t type yet
-  if (!is_win) {
+  if (is_clang && !is_win) {
     source_set("libvpx_intrinsics_sve") {
       check_includes = false
       configs -= [ "//build/config/compiler:chromium_code" ]
@@ -568,7 +568,7 @@
   if (current_cpu == "arm64" || current_cpu == "arm64e") {
     deps += [ ":libvpx_intrinsics_neon_dotprod" ]
     deps += [ ":libvpx_intrinsics_neon_i8mm" ]
-    if (!is_win) {
+    if (is_clang && !is_win) {
       deps += [
         ":libvpx_intrinsics_sve",
         ":libvpx_intrinsics_sve2",
diff --git a/core/dep/_google-webrtc/src/third_party/libaom/BUILD.gn b/core/dep/_google-webrtc/src/third_party/libaom/BUILD.gn
--- a/core/dep/_google-webrtc/src/third_party/libaom/BUILD.gn
+++ b/core/dep/_google-webrtc/src/third_party/libaom/BUILD.gn
@@ -265,7 +265,7 @@
   # There are known problems with LLVM-based compilers targeting Windows for
   # SVE code generation. Since there are no client Windows devices that
   # support SVE(2) at this time, disable SVE(2) on AArch64 Windows targets.
-  if (!is_win) {
+  if (is_clang && !is_win) {
     source_set("libaom_intrinsics_sve") {
       check_includes = false
       configs -= [ "//build/config/compiler:chromium_code" ]
@@ -351,7 +351,7 @@
       ":libaom_intrinsics_neon_dotprod",
       ":libaom_intrinsics_neon_i8mm",
     ]
-    if (!is_win) {
+    if (is_clang && !is_win) {
       deps += [
         ":libaom_intrinsics_sve",
         ":libaom_intrinsics_sve2",
diff --git a/core/dep/_google-webrtc/src/rtc_base/platform_thread_types.cc b/core/dep/_google-webrtc/src/rtc_base/platform_thread_types.cc
index 45bf7a1..3a36ef7 100644
--- a/core/dep/_google-webrtc/src/rtc_base/platform_thread_types.cc
+++ b/core/dep/_google-webrtc/src/rtc_base/platform_thread_types.cc
@@ -16,7 +16,7 @@
 #include <sys/prctl.h>
 #include <sys/syscall.h>
 
-#if !defined(WEBRTC_ARCH_ARM) && !defined(WEBRTC_ARCH_ARM64)
+#if !defined(WEBRTC_ARCH_ARM) && !defined(WEBRTC_ARCH_ARM64) && !defined(__riscv)
 #include <asm/unistd_64.h>
 #endif
 #endif
